addons/binding/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/messages/RamRequestPayload.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.paradoxalarm.internal.communication.messages;
+
+import org.openhab.binding.paradoxalarm.internal.exceptions.ParadoxBindingException;
+import org.openhab.binding.paradoxalarm.internal.util.ParadoxUtil;
+
+/**
+ * The {@link RamRequestPayload} Object representing payload of IP packet which retrieves data from Paradox RAM
+ *
+ * @author Konstantin_Polihronov - Initial contribution
+ */
+public class RamRequestPayload extends MemoryRequestPayload implements IPPacketPayload {
+
+    public RamRequestPayload(int address, byte bytesToRead) throws ParadoxBindingException {
+        super(address, bytesToRead);
+    }
+
+    @Override
+    protected byte calculateControlByte() {
+        byte byteValue = (byte) 0;
+        return ParadoxUtil.setBit(byteValue, 7, 1);
```

Comment:
This looks like it's a constant. So you might just create a static field of it so it's not calculated every time. For example:
`private static final byte CONTROL_BYTE = ParadoxUtil.setBit((byte) 0, 7, 1);`
See if this pattern can be applied in other places in your code.

Reply:
The method will return static pre-calculated constatn in this class. Thanks !